thingooConnector/connector.py

```diff
@@ -121,11 +123,12 @@ def api_request(self, method, endpoint, data=None):
             "Authorization": "Bearer " + self._token.access_token(),
             "Content-Type": "application/json"
         }
-        response = requests.request(method=method, url=url, json=data, headers=headers)
+        data = json.dumps(data, cls=ComplexEncoder)
+        response = requests.request(method=method, url=url, data=data, headers=headers)
         if response.status_code == HTTPStatus.UNAUTHORIZED:
             # Token expired, update token and try again
             self._update_token()
-            response = requests.request(method=method, url=url, json=data, headers=headers)
+            response = requests.request(method=method, url=url, data=data, headers=headers)
```

Comment:
Duplicated lines 127 and 131, can we avoid the duplication or at least extract it to another method?